src/csharp/Microsoft.Spark/Sql/SparkSession.cs

```diff
@@ -39,6 +40,10 @@ internal SparkSession(JvmObjectReference jvmObject)
                     (JvmObjectReference)_jvmObject.Invoke("sparkContext")));
             _catalog = new Lazy<Catalog.Catalog>(
                 () => new Catalog.Catalog((JvmObjectReference)_jvmObject.Invoke("catalog")));
+
+            _version = new Lazy<Version>(() => new Version((string)_jvmObject.Jvm.CallStaticJavaMethod(
+                "org.apache.spark.deploy.dotnet.DotnetRunner",
+                "SPARK_VERSION")));
```

Comment:
I don't think we need to cache this.

Reply:
Agreed. However, believe calling it once via the constructor and saving it to a private field is still best via querying the JVM every time.